File: ruby/lib/Sikaru/eval_seeds/types/create_eval_seed_request.rb
# frozen_string_literal: true

module Sikaru
  module EvalSeeds
    module Types
      class CreateEvalSeedRequest < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :dataset_name, -> { String }, optional: true, nullable: false, api_name: "datasetName"

        field :evaluator_name, -> { String }, optional: true, nullable: false, api_name: "evaluatorName"

        field :issue_id, -> { String }, optional: false, nullable: false, api_name: "issueId"

        field :issue_title, -> { String }, optional: false, nullable: false, api_name: "issueTitle"

        field :trace_ids, -> { Internal::Types::Array[String] }, optional: false, nullable: false, api_name: "traceIds"
      end
    end
  end
end

File: typescript/src/api/resources/evalSeeds/client/requests/CreateEvalSeedRequest.ts

/**
 * @example
 *     {
 *         issueId: "issueId",
 *         issueTitle: "issueTitle",
 *         traceIds: ["traceIds"]
 *     }
 */
export interface CreateEvalSeedRequest {
    datasetName?: string | null;
    evaluatorName?: string | null;
    issueId: string;
    issueTitle: string;
    traceIds: string[];
}
